Add auth_mode to azure_rm_storageblob

ephracis opened the pull request:
Add the parameter `auth_mode` to the
`azure_rm_storageblob` module, making it possible to create blobs without using access keys.

##### SUMMARY
Add the parameter `auth_mode` to the `azure_rm_storageblob` module.

Fixes #1283, #1255 

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Feature Pull Request

##### COMPONENT NAME
azure_rm_storageblob

##### ADDITIONAL INFORMATION
This allows customer which has disabled access keys to use the module.

ephracis committed:
Add auth_mode to azure_rm_storageblob

Add the parameter `auth_mode` to the
`azure_rm_storageblob` module, making it possible to create blobs
without using access keys.

Fred-sun commented on plugins/modules/azure_rm_storageblob.py:
```diff
@@ -214,13 +225,19 @@
 
 from ansible_collections.azure.azcollection.plugins.module_utils.azure_rm_common import AzureRMModuleBase
-
+from ansible.module_utils.basic import env_fallback
```
Expected two lines!
```suggestion
from ansible.module_utils.basic import env_fallback

```

Fred-sun commented on plugins/module_utils/azure_rm_common.py:
```diff
@@ -675,19 +675,23 @@ def check_provisioning_state(self, azure_object, requested_state='present'):
                     azure_object.name, azure_object.provisioning_state, AZURE_SUCCESS_STATE))
 
-    def get_blob_service_client(self, resource_group_name, storage_account_name):
+    def get_blob_service_client(self, resource_group_name, storage_account_name, auth_mode = 'key'):
```
unexpected spaces, unexpected spaces around keyword / parameter equals

```suggestion
    def get_blob_service_client(self, resource_group_name, storage_account_name, auth_mode='key'):
```

ephracis committed:
Fix sanity tests

ephracis committed:
Add integration tests

Test using auth_mode='login' for storage blobs.
This has been tested on a machine with MSI.

ephracis commented:
Fixed sanity tests and adjusted integration tests.

Not sure how this will work in your pipeline but I have tried it on a server where we are using `ANSIBLE_AZURE_AUTH_SOURCE: msi` against an Azure account where access keys are forbidden and it works fine, which it doesn't without this change.

Hope that's enough for now. :)

ephracis committed:
Remove spaces around equal sign

xuzhang3 commented:
@ephracis LGTM
